src/redux/slices/ensResolver.slice.ts

```diff
@@ -0,0 +1,114 @@
+import urlcat from 'urlcat'
+import { ethers } from "ethers";
+import { createApi,  fakeBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export interface NameInfo {
+  rnsName: string
+  ensName: string | null
+  address: string
+}
+
+export interface ProfileInfo {
+  avatar: string[]
+  bio: string
+  name: string
+}
+
+interface RSS3Info {
+  profile: ProfileInfo
+}
+
+
+export const getNameById = async (id: string ) => {
+  if (!id) return ''
+    const url = urlcat('https://rss3.domains/address/:id', { id })
+    const rsp = (await (await fetch(url)).json()) as NameInfo
+    return rsp.ensName
+}
+
+export const getAddressByName = async (id: string) => {
+  if (!id) return ''
+  const url = urlcat('https://rss3.domains/name/:id', { id })
+  const rsp = (await (await fetch(url)).json()) as NameInfo
+  return rsp.address
+}
+
+
+export const getUserAvatar = async (address: string) => {
+  if (!address) return ''
+  const url = urlcat('https://hub.pass3.me/:address', { address })
+  const rsp = (await (await fetch(url)).json()) as  RSS3Info
+  return rsp.profile
+
+}
+
+
+export const ensApi = createApi({
+  reducerPath: "ens",
+  baseQuery: fakeBaseQuery(),
+  endpoints: (builder) => {
+
+    const mainnetProvider = new ethers.providers.InfuraProvider(
+      "mainnet",
+      "c44fac7726a64d5bbbb3a1c51f02d75b"
+    );
+
+    return {
+      resolveName: builder.query<
+        { address: string; name: string } | null,
+        string
+      >({
+        queryFn: async (name) => {
+          if (ethers.utils.isAddress(name)) {
+            return { data: null };
+          }
+
+          if (!name.endsWith('.eth')) {
```

Comment:
This approach is flawed: _"The ENS namespace includes both .eth names (which are native to ENS) and DNS names imported into ENS. Because the DNS suffix namespace expands over time, a hardcoded list of name suffixes for recognizing ENS names will regularly be out of date, leading to your application not recognizing all valid ENS names. To remain future-proof, a correct integration of ENS treats any dot-separated name as a potential ENS name and will attempt a look-up."_ (source: https://docs.ens.domains/dapp-developer-guide/resolving-names)

We can do an early return when there's no "." in the given name but ".eth" is too specific.